### PR TITLE
New exception columns

### PR DESCRIPTION
**Pull request by pragnagopa**
Add new columns for splitting the error information from details column into exceptionType, exceptionMessage and exceptionStackTrace columns. This will make it easier to perform aggregate queries over errors in function logs

### CODE REVIEW

File: src/WebJobs.Script.WebHost/Diagnostics/SystemTraceWriter.cs [hunk 1]
```diff
@@ -71,9 +75,32 @@ public override void Trace(TraceEvent traceEvent)
...
             {
                 details = Sanitizer.Sanitize(traceEvent.Exception.ToFormattedString());
+                if (
+                    (traceEvent.Exception is FunctionInvocationException ||
+                    traceEvent.Exception is AggregateException ||
+                    traceEvent.Exception is TargetInvocationException) &&
+                    traceEvent.Exception.InnerException != null)
+                {
+                    GetExceptionDetails(traceEvent.Exception.InnerException, out exceptionType, out exceptionMessage, out exceptionStackTrace);
+                }
```
Comment (paulbatum) [inline src/WebJobs.Script.WebHost/Diagnostics/SystemTraceWriter.cs]: I'm wondering whether we should just make it a rule that we want the exception columns to be populated with the inner most exception, regardless of type?

File: src/WebJobs.Script.WebHost/Diagnostics/SystemTraceWriter.cs [hunk 2]
```diff
@@ -71,9 +75,32 @@ public override void Trace(TraceEvent traceEvent)
...
+
+        private void GetExceptionDetails(Exception exception, out string exceptionType, out string exceptionMessage, out string exceptionStackTrace)
+        {
+            if (exception == null)
+            {
+                throw new ArgumentNullException(nameof(exception));
+            }
+            exceptionType = exception.GetType().ToString();
+            exceptionMessage = exception.Message ?? string.Empty;
+            exceptionStackTrace = Sanitizer.Sanitize(exception.ToFormattedString()) ?? string.Empty;
```
Comment (paulbatum) [inline src/WebJobs.Script.WebHost/Diagnostics/SystemTraceWriter.cs]: For the stack trace, is this better than `exception.StackTrace`?

File: src/WebJobs.Script.WebHost/Diagnostics/EventGenerator.cs [hunk 3]
```diff
@@ -270,7 +318,7 @@ public class FunctionsSystemLogsEventSource : ExtendedEventSource
         {
             internal static readonly FunctionsSystemLogsEventSource Instance = new FunctionsSystemLogsEventSource();
 
-            [Event(65520, Level = EventLevel.Verbose, Channel = EventChannel.Operational, Version = 1)]
+            [Event(65520, Level = EventLevel.Verbose, Channel = EventChannel.Operational, Version = 2)]
```
Comment (paulbatum) [inline src/WebJobs.Script.WebHost/Diagnostics/EventGenerator.cs]: I'm curious how this versioning works? Have you tried it out before?

Comment (pragnagopa) [reply src/WebJobs.Script.WebHost/Diagnostics/EventGenerator.cs]: Yes. Version needs to be incremented any time we add columns. MDS had documentation regarding this.

File: src/WebJobs.Script.WebHost/Diagnostics/SystemTraceWriter.cs (hunk 2 above)
Comment (brettsam) [reply src/WebJobs.Script.WebHost/Diagnostics/SystemTraceWriter.cs]: Also, we should be sanitizing the message. I don't think the stack trace needs to be sanitized.

File: src/WebJobs.Script.WebHost/Diagnostics/EventGenerator.cs (hunk 3 above)
Comment (brettsam) [reply src/WebJobs.Script.WebHost/Diagnostics/EventGenerator.cs]: So if you deploy this with a private Site Extension today, will we see this show up in Kusto for your app?

Comment (pragnagopa) [reply src/WebJobs.Script.WebHost/Diagnostics/EventGenerator.cs]: Yes. We can test this in Kusto!!

File: src/WebJobs.Script.WebHost/Diagnostics/SystemTraceWriter.cs (hunk 1 above)
Comment (pragnagopa) [reply src/WebJobs.Script.WebHost/Diagnostics/SystemTraceWriter.cs]: Will update to look for the inner most exception.

File: src/WebJobs.Script.WebHost/Diagnostics/SystemTraceWriter.cs (hunk 2 above)
Comment (pragnagopa) [reply src/WebJobs.Script.WebHost/Diagnostics/SystemTraceWriter.cs]: Ok. Thanks. Will update.

**Pull request merged**